test/unit/entrypoint/test_file_permissions.py

```diff
@@ -12,26 +13,28 @@ def test_file_inspector_non_existing_file(mocker):
     mocker.patch("os.stat")
     # need to mock os.path.exists as os.path.exists seems to call os.stat :)
     mocker.patch("os.path.exists", return_value=False)
-    testee = entrypoint.FileInspector("/non/existing/file")
+    testee = entrypoint.FileInspector(Path("/non/existing/file"))
     actual = testee.is_group_accessible()
     assert actual == False
     assert not os.stat.called
 
 
+def test_file_inspector_group_id_non_existing_file():
+    testee = entrypoint.FileInspector(Path("/non/existing/file"))
+    with pytest.raises(FileNotFoundError):
+        testee.group_id
+
+
 def test_file_inspector_group_accessible(accessible_file):
     testee = entrypoint.FileInspector(accessible_file)
     assert testee.is_group_accessible()
 
 
-def test_file_inspector_not_group_accessible(non_accessible_file, caplog):
+def test_file_inspector_not_group_accessible(non_accessible_file):
     testee = entrypoint.FileInspector(non_accessible_file)
-    assert not testee.is_group_accessible()
-    assert re.match(r"ERROR .* No rw permissions for group", caplog.text)
-
-
-def test_group_access_unknown_group_id():
-    testee = entrypoint.GroupAccess(None, None)
-    assert testee._find_group_name(9999999) is None
+    with pytest.raises(PermissionError) as err:
+        testee.is_group_accessible()
+    assert re.match(r"No rw permissions for group", str(err.value))
```

Comment:
this you can add directly to the pytest.raises(ExceptionType, match)

Reply:
Ah, thanks - see next push